src/main/java/com/autilus/hw0706/ListImpl.java

```diff
@@ -0,0 +1,110 @@
+package com.autilus.hw0706;
+
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+public class ListImpl<T> implements List<T> {
+    private int INITIAL_SIZE = 10;
+    private T[] values = (T[]) new Object[INITIAL_SIZE];
+    private int size = 0;
+
+    private void increaseArray() {
+        values = Arrays.copyOf(values, (values.length * 3) / 2);
+    }
+
+    private void checkIndexSize(int index) {
+        if (index < 0 || index > size) {
+            throw new NoSuchElementException();
+        }
+    }
+
+    @Override
+    public void add(T value) {
+        if (size == values.length) {
+            increaseArray();
+        }
+        values[size++] = value;
+    }
+
+    @Override
+    public void add(int index, T value) {
+        if (size == index) {
+            add(value);
+            return;
+        } else {
```

Comment:
we can remove this else-block

Reply:
we can't! Because if _size == index_ and after it we use checkIndexSize method. Then we have an exception. because of logic of checkIndexSize method